### ISSUE TITLE
Error importing images

### ISSUE THREAD

**Issue by Tor-Saxberg** [ogkalu2/comic-translate#14]
Whenever I try to import an image the gui crashes and I get this error:
python comic.py
[1]    72617 trace trap python comic.py

the error code changes each time.
I swapped to dearpygui==1.10.0 since pip didn't find a 1.10.1 package. 
I'm on mac 14.2

**Comment by ogkalu2** [comment]
Does the dialog box open at all ? 

check what versions of you can install with:
pip install dearpygui==

Then try uninstalling dearpygui and installing one of the older versions

**Comment by KakoitoTatar** [comment]
Dialog box don't open at all, if you try second time app crashes
`Fatal Python error: PyEval_SaveThread: the function must be called with the GIL held, but the GIL is released (the current Python thread state is NULL)
Python runtime state: initialized

Current thread 0x00000002bba07000 (most recent call first):
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/tkinter/commondialog.py", line 46 in show
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/tkinter/filedialog.py", line 400 in askopenfilenames
  File "/Users/kakoitotatar/Projects/Pet projects/comic-translate/app/callbacks.py", line 27 in import_images
  File "/Users/kakoitotatar/Projects/Pet projects/comic-translate/comic.py", line 85 in <lambda>
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/tkinter/__init__.py", line 2270 in __init__
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/tkinter/__init__.py", line 298 in _get_default_root
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/tkinter/__init__.py", line 2533 in _setup
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/tkinter/__init__.py", line 2566 in __init__
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/tkinter/__init__.py", line 3124 in __init__
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/tkinter/commondialog.py", line 42 in show
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/tkinter/filedialog.py", line 400 in askopenfilenames
  File "/Users/kakoitotatar/Projects/Pet projects/comic-translate/app/callbacks.py", line 27 in import_images
  File "/Users/kakoitotatar/Projects/Pet projects/comic-translate/comic.py", line 85 in <lambda>

Thread 0x00000001e1985c40 (most recent call first):
  File "/Users/kakoitotatar/Projects/Pet projects/comic-translate/venv/lib/python3.9/site-packages/dearpygui/dearpygui.py", line 102 in start_dearpygui
  File "/Users/kakoitotatar/Projects/Pet projects/comic-translate/comic.py", line 348 in <module>
[1]    51456 abort      python comic.py
`

**Comment by Tor-Saxberg** [comment]
There's no Dialog box and no further error message. I'm not sure where @KakoitoTatar got the Fatal Python error from; maybe it's a different issue or platform. 

with dearpygui==1.5.1 I get an additional error: 
Glfw Error 65544: Cocoa: Failed to find service port for display
but this seems to be resolved with dearpygui==1.6.0 and later.

**Comment by ogkalu2** [comment]
@Tor-Saxberg @KakoitoTatar 

Try running this in a virtual environment

python -m venv comic-venv
then:
source comic-venv/bin/activate ( for mac)
comic-venv\Scripts\activate (for windows)

Now go to the directory of the comic folder:
cd /path/to/comic-translate folder

pip install -r requirements.txt
python comic.py

**Comment by Tor-Saxberg** [comment]
That's what I did the first time. Only packages and dependencies are in the venv (result of `pip install -r requirements.txt` ), though I did have to `brew install python-tk@3.10` to get tkinter support. does this work for you without error in a blank venv? with the latest version of pip, you should at least face the missing `dearpygui==1.10.1` error in a clean setup.

**Comment by ogkalu2** [comment]
I see. 

All Import does is call a tkinter file dialoguebox so I'm really not sure what the issue is here.

It works for me in a black venv but I use Windows.

**Comment by ogkalu2** [comment]
I'll try to alter the code to run the dialogue box manually and push an update later today.

**Comment by ogkalu2** [comment]
Okay. No more tkinter for MacOS. Can you try it now ?

**Comment by Tor-Saxberg** [comment]
A pop-up loads now saying "nothing was imported", but there was no window to select any files.

**Comment by ogkalu2** [comment]
Try again now. also see if multiple selections work for images (assuming single selection works)

**Comment by Tor-Saxberg** [comment]
Now the pop-up says "Imported!" but there was still no file selection window.

**Comment by ogkalu2** [comment]
Run this on script editor and let me know what happens

-- apath - default path for dialogs to open to
-- cmd   - "Select", "Save"
-- fileTypes - list of allowed file extensions
-- allowMultiple - boolean to allow multiple file selections
on run argv
    set userCanceled to false
    if (count of argv) < 3 then
        tell application "System Events" to display dialog "Not enough arguments" ¬
            giving up after 10
    else
        set apath to POSIX file (item 1 of argv) as alias
        set action to (item 2 of argv) as text
        set fileTypes to {".png",".jpg",".jpeg",".webp",".bmp"}
        set allowMultiple to true
    end if
    try
    if action contains "Select" then
        set fpath to POSIX path of (choose file default location apath ¬
                    of type fileTypes without invisibles, multiple selections allowed allowMultiple and ¬
                    showing package contents)
    else if action contains "Save" then
        set fpath to POSIX path of (choose file name default location apath)
    end if
    return fpath as text
    on error number -128
        set userCanceled to true
    end try
    if userCanceled then
        return "Cancel"
    else
        return fpath
    end if
end run

**Comment by ogkalu2** [comment]
Run this on script editor and let me know what happens

```
-- apath - default path for dialogs to open to
-- cmd   - "Select", "Save"
-- fileTypes - list of allowed file extensions
-- allowMultiple - boolean to allow multiple file selections
on run argv
    set userCanceled to false
    if (count of argv) < 3 then
        tell application "System Events" to display dialog "Not enough arguments" ¬
            giving up after 10
    else
        set apath to POSIX file (item 1 of argv) as alias
        set action to (item 2 of argv) as text
        set fileTypes to {file_types_str}
        set allowMultiple to {multiple_selections}
    end if
    try
    if action contains "Select" then
        set fpath to POSIX path of (choose file default location apath ¬
                    of type fileTypes without invisibles, multiple selections allowed allowMultiple and ¬
                    showing package contents)
    else if action contains "Save" then
        set fpath to POSIX path of (choose file name default location apath)
    end if
    return fpath as text
    on error number -128
        set userCanceled to true
    end try
    if userCanceled then
        return "Cancel"
    else
        return fpath
    end if
end run
```

**Comment by Tor-Saxberg** [comment]
I tried to compiled this in mac script editor, but it gave me a compilation error: 
Syntax Error: Expected “given”, “with”, “without”, other parameter name, etc. but found identifier.

it highlighted "allowMultiple" in this section:
if action contains "Select" then
        set fpath to POSIX path of (choose file default location apath ¬
                    of type fileTypes without invisibles, multiple selections allowed allowMultiple and ¬
                    showing package contents)

**Comment by ogkalu2** [comment]
OK. replace that block of code with:
```
if action contains "Select" then
    set fpath to POSIX path of (choose file default location apath ¬
                of type fileTypes without invisibles, multiple selections allowed and ¬
                showing package contents)
```
and try again

>I've not written mac scripts before, so just to be sure: are the first four lines part of the script?
They are commented lines. -- is the applescript equivalent of # for python

**Comment by Tor-Saxberg** [comment]
I got two messages: "Not enough arguments" and then "Script Error: The variable action is not defined" highlighting "action" in 
```
try
		if action contains "Select" then
		```

**Comment by ogkalu2** [comment]
my bad. run this instead. argv would have been the variables the script would receive from python.

```
set userCanceled to false
set apath to POSIX file "~/Desktop" as alias
set action to "Select"
set fileTypes to {".ttf", ".ttc", ".otf", ".woff", ".woff2"}
set allowMultiple to true

try
if (action contains "Select") and (allowMultiple) then
    set fpath to POSIX path of (choose file default location apath ¬
                of type fileTypes without invisibles, multiple selections allowed and ¬
                showing package contents) 
else if action contains "Select" then
    set fpath to POSIX path of (choose file default location apath ¬
                of type fileTypes without invisibles and ¬
                showing package contents)
else if action contains "Save" then
    set fpath to POSIX path of (choose file name default location apath)
end if
return fpath as text
on error number -128
    set userCanceled to true
end try
if userCanceled then
    return "Cancel"
else
    return fpath
end if
```

**Comment by Tor-Saxberg** [comment]
"Scripte Error: Can’t make file ":~:Desktop" into type alias."
`error "Can’t make file \":~:Desktop\" into type alias." number -1700 from file ":~:Desktop" to alias`

**Comment by ogkalu2** [comment]
in this line:

set apath to POSIX file "~/Desktop" as alias

replace "~/Desktop" with the path to your desktop folder

**Comment by Tor-Saxberg** [comment]
Do you mean the absolute file path? `cd ~/Desktop` works in terminal. 
A quick google search suggests relative paths (~) don't work in apple scripts.
I tried `(path to desktop)` instead, but I get this error:
`error "Can’t make file \":Macintosh HD/Users/username/Desktop/\" into type alias." number -1700 from file ":Macintosh HD/Users/username/Desktop/" to alias`

**Comment by ogkalu2** [comment]
>it seems that "/Users/username/Desktop" worked; it opened a file selector window.

Ok that's great. let's see if selections work. select a single font file and click choose and see what the output is then run the script again, select multiple files and see what the output is. 

also this should constrain what files you can choose to font files. is that working properly ?

**Comment by ogkalu2** [comment]
in the comic-translate folder, there is a font folder

**Comment by Tor-Saxberg** [comment]
It let's me select a font
`"/Users/username/Desktop/comic-translate-main/fonts/NotoSansJP-Medium.ttf"`
but the window ui it doesn't let me select multiple fonts

**Comment by Tor-Saxberg** [comment]
GPT provided this code which does work:
```
set userCanceled to false
set apath to POSIX file "/Users/torsaxberg/Desktop" as alias
set action to "Select"
set fileTypes to {".ttf", ".ttc", ".otf", ".woff", ".woff2"}
set allowMultiple to true

try
	if (action contains "Select") and (allowMultiple) then
		set fpaths to choose file default location apath ¬
			of type fileTypes with multiple selections allowed and showing package contents without invisibles
		set selectedFiles to {}
		repeat with aFile in fpaths
			set end of selectedFiles to POSIX path of aFile
		end repeat
	else if action contains "Select" then
		set fpath to POSIX path of ¬
			(choose file default location apath ¬
				of type fileTypes without invisibles and showing package contents)
	else if action contains "Save" then
		set fpath to POSIX path of (choose file name default location apath)
	end if
	if (count of selectedFiles) is greater than 0 then
		return selectedFiles
	else
		return fpath
	end if
on error number -128
	set userCanceled to true
end try

if userCanceled then
	return "Cancel"
else
	return fpath
end if
```

**Comment by ogkalu2** [comment]
in the original code, replace:
```
of type fileTypes without invisibles with multiple selections allowed and ¬
showing package contents) 
```
with
of type fileTypes with multiple selections allowed and showing package contents without invisibles

and run it again. I need to see the output of that to play nice with python

**Comment by Tor-Saxberg** [comment]
it works for a single file, but gives an error for multiple.
the GPT explanation says that you need to handle a list when multiple files are selected.

**Comment by ogkalu2** [comment]
ok. what's the output of GPT's code ?

**Comment by Tor-Saxberg** [comment]
`{"/Users/username/Desktop/comic-translate-main/fonts/NotoSansJP-Medium.ttf", "/Users/username/Desktop/comic-translate-main/fonts/NotoSansKR-Medium.ttf"}`

it works for any file types (including mixed file types).

**Comment by ogkalu2** [comment]
replace this: 
set fileTypes to {".ttf", ".ttc", ".otf", ".woff", ".woff2"}
with
set fileTypes to {"public.font"}

**Comment by Tor-Saxberg** [comment]
GPT says that 

> The of type fileTypes parameter in the choose file dialog specifies the acceptable file types that will be displayed to the user, but it does not enforce a strict filter on the selected files. It means that the dialog will show only the files with the specified extensions in the file chooser window, but it doesn't prevent users from manually selecting files with different extensions if they navigate to a different directory.

**Comment by ogkalu2** [comment]
Ok. try the latest pull and see if it finally works.

**Comment by Tor-Saxberg** [comment]
There's still no pop-up window.

**Comment by ogkalu2** [comment]
Try again

**Comment by Tor-Saxberg** [comment]
same result

**Comment by ogkalu2** [comment]
Yeah try again. If nothing else, we should get the error this time

**Comment by Tor-Saxberg** [comment]
830:835: syntax error: Expected end of line but found plural class name. (-2741)
Python error: [1]

**Comment by Tor-Saxberg** [comment]
I had GPT isolate the applescript for me. it worked perfectly, so I don't understand the issue. could it be with the python injections like  {file_types_str}? I'm not sure what the 830:835 location points to. 
```
-- Set the default path for the dialog to open to
set apath to POSIX file "/Users/yourusername/Desktop" as alias

-- Set the action to "Select" and define allowed file types
set action to "Select"
set fileTypes to {"ttf", "ttc", "otf", "woff", "woff2"}

-- Set whether to allow multiple file selections
set allowMultiple to true

-- Initialize variable to hold the selected file paths
set selectedFiles to {}

try
    -- Check if the action is to select files
    if action contains "Select" then
        -- Handle multiple file selections
        if allowMultiple then
            set fpaths to choose file with prompt "Select files:" default location apath of type fileTypes with multiple selections allowed and showing package contents without invisibles
            -- Loop through the selected files to convert them to POSIX paths
            repeat with aFile in fpaths
                set end of selectedFiles to POSIX path of aFile
            end repeat
            -- Combine the POSIX paths into a single string to return
            set AppleScript's text item delimiters to ", "
            set fpath to selectedFiles as text
            set AppleScript's text item delimiters to ""
        else
            -- Handle single file selection
            set fpath to POSIX path of (choose file with prompt "Select a file:" default location apath of type fileTypes without invisibles and showing package contents)
        end if
    end if
on error number -128
    -- Handle user cancel action
    return "Cancel"
end try

-- Return the selected file path(s) or indicate cancellation
if selectedFiles is not {} then
    return fpath
else
    return "No files selected or action cancelled."
end if

```

**Comment by Tor-Saxberg** [comment]
i noticed that apath is not injected like the other python strings. is that intentional? i also don't understand how argv is populated.

**Comment by ogkalu2** [comment]
i've done away with argv. see if that works

**Comment by Tor-Saxberg** [comment]
wait I may have used an old script. let me try again

**Comment by Tor-Saxberg** [comment]
/Users/torsaxberg/Desktop
{}
Image files
{"png", "jpg", "jpeg", "webp", "bmp"}
230:235: execution error: The variable Users is not defined. (-2753)
Python error: [1]

it looks like the apath may not have quotes around it

**Comment by ogkalu2** [comment]
add quotes to cmd too - "{cmd}". though i just updated to reflect that.

**Comment by Tor-Saxberg** [comment]
it works!

**Comment by ogkalu2** [comment]
It should be in the same folder as where the original image/file was? Do you see anything ?

also do multiple selections also work ?

**Comment by Tor-Saxberg** [comment]
it says comic has been translated, but nothing is showing up in the same directory as the jpg files. I'll create a separate issue

**Comment by ogkalu2** [comment]
Okay. when it runs. does it show the correct number of images ? e.g image 1 of 5 if you selected say 5 images ?

run it again but tick one of the preview options and see if it previews
